addons/website_sale/data/demo.xml

```diff
@@ -643,19 +642,26 @@
             <field name="uom_id" ref="uom.product_uom_unit"/>
             <field name="uom_po_id" ref="uom.product_uom_unit"/>
             <field name="description_sale">Warranty, issued to the purchaser of an article by its manufacturer, promising to repair or replace it if necessary within a specified period of time.</field>
-            <field name="attribute_value_ids" eval="[(6,0,[ref('product.product_attribute_value_5')])]"/>
             <field name="categ_id" ref="product.product_category_3"/>
             <field name="invoice_policy">delivery</field>
+            <field name="public_categ_ids" eval="[(6,0,[ref('website_sale.services')])]"/>
         </record>
-        <record id="product_template_attribute_line_1" model="product.template.attribute.line">
-            <field name="product_tmpl_id" ref="product_product_1_product_template"/>
+
+        <record id="website_sale.product_template_attribute_line_1" model="product.template.attribute.line">
+            <field name="product_tmpl_id" ref="website_sale.product_product_1_product_template"/>
             <field name="attribute_id" ref="product.product_attribute_3"/>
             <field name="value_ids" eval="[(6,0,[ref('product.product_attribute_value_5'), ref('product.product_attribute_value_6')])]"/>
         </record>
-        <record id="product_product_1_product_template" model="product.template">
-            <field name="attribute_line_ids" eval="[(6,0,[ref('product_template_attribute_line_1')])]"/>
-            <field name="public_categ_ids" eval="[(6,0,[ref('website_sale.services')])]"/>
+
+        <record id="website_sale.product_product_1" model="product.product">
+            <field name="default_code">SERV_125889</field>
+            <field name="product_tmpl_id" ref="website_sale.product_product_1_product_template"/>
+            <field name="attribute_value_ids" eval="[(6,0,[ref('product.product_attribute_value_5')])]"/>
         </record>
+        <!-- clean up/generate missing variants -->
+        <function model="product.template" name="create_variant_ids">
+            <value model="product.template" eval="ref('website_sale.product_product_1_product_template')"/>
```

Comment:
eval in the function tag should work

Reply:
Ok thanks, I just moved/copied this from elsewhere, but if it can be cleaned I'll try.